bestbook/views/__init__.py

```diff
@@ -257,6 +257,23 @@ def get(self, olid):
             Observation.book_id == book.id).all()]
         }
 
+    def delete(self, olid):
+        try:
+            book = Book.get(edition_olid=olid) if 'M' in olid else Book.get(work_olid=olid)
+        except RexException:
+            return "Not Found", 404
+        # TODO: Change this to fetch username using request header S3 keys:
+        username = request.args.get('username')
+
+        if username:
+            observations = Observation.get_all(book_id=book.id,
+                                               username=username)
+            for o in observations:
+                o.remove()
+            return "OK", 200
+        else:
+            return "Bad Request", 400
```

Comment:
Fast fail on no username...
```suggestion
        username = request.args.get('username')
        if not username:
            return "Bad Request", 400

        for observation in Observation.get_all(book_id=book.id,
                                               username=username):
            observation.remove()
        return "OK", 200
```

Reply:
That makes sense, but I'm going to hold off on making any changes to this now.  In target state, username will be fetched by using S3 keys that will be sent in the request header.